#!/usr/bin/env python
""" Script to quickly investigate failed estimation runs.
"""
# standard library
import numpy as np
import sys
import os

# virtual environment
if not hasattr(sys, 'real_prefix'):
    raise AssertionError('Please use a virtual environment for testing')

# project library
sys.path.insert(0, os.environ['ROBUPY'])


# project library
from modules.auxiliary import *

''' Request
'''
# Failed seeds ...
#91317, 61905, 14397, 39101, 11692, 62692

label, seed = '98', 28126

''' Error Reproduction
'''

if True:


    np.random.seed(seed)

    try:
        import modules.battery as Tests
        test = getattr(Tests, 'test_' + label)
    except AttributeError:
        from robupy.tests.tests import Tests
        test = getattr(Tests, 'test_' + label)

    count = 0
    while True:
        print('investigate ', count)
        test()
        count = count + 1

        if count == 1:
            break
